File: 2657-find-the-prefix-common-array-of-two-arrays/2657-find-the-prefix-common-array-of-two-arrays.py
class Solution:
    def findThePrefixCommonArray(self, A: List[int], B: List[int]) -> List[int]:

        count = 0
        array = []

        my_dict = dict()

        for i in range(len(A)):
            #A
            if A[i] in my_dict:
                my_dict[A[i]] += 1
            else:
                my_dict[A[i]] = 1
            
            #B
            if B[i] in my_dict:
                my_dict[B[i]] += 1
            else:
                my_dict[B[i]] = 1
            
            #Check dict for size above 2
            for value in my_dict.values():
                if value >= 2:
                    count += 1
            
            array.append(count)
            count = 0

        return array
                


            
            
            
        

File: 2657-find-the-prefix-common-array-of-two-arrays/2657-find-the-prefix-common-array-of-two-arrays.cpp
class Solution {
public:
    vector<int> findThePrefixCommonArray(vector<int>& A, vector<int>& B) 
    {
        vector<int> x(A.size(), 0);
        unordered_set<int> visited;

        int count = 0;

        for (int i{}; i < B.size(); i++)
        {
            res[B[i]] += 1;
            res[A[i]] += 1;

            if (res[B[i]] >= 2 && visited.find(B[i]) == visited.end())
            {
                count += 1;
                visited.insert(B[i]);
            }

            if (res[A[i]] >= 2 && visited.find(A[i]) == visited.end())
            {
                count += 1;
                visited.insert(A[i]);
            }

            //Add it to vec
            x[i] = count;
        }


        return x;
    }

private:
    unordered_map<int, int> res;
};